File: 2078-two-furthest-houses-with-different-colors/2078-two-furthest-houses-with-different-colors.cpp
class Solution {
public:
    int maxDistance(vector<int>& c) {
     int n=c.size();
     int maxi=0;
     for(int i=0;i<n;++i)
     {
         if(c[i]!=c[0])
            maxi=max(maxi,i);
     }
    for(int i=n-2;i>=0;--i)
     {
         if(c[i]!=c[n-1])
            maxi=max(maxi,n-1-i);
     }
    return maxi;
    }
};

File: 2078-two-furthest-houses-with-different-colors/2078-two-furthest-houses-with-different-colors.java
class Solution {

    public int maxDistance(int[] colors) {
        int n = colors.length;
        int res = 0; // the maximum distance between two houses of different colors
        // traverse the indices of two houses and maintain the maximum distance
        for (int i = 0; i < n; i++) {
            for (int j = i + 1; j < n; j++) {
                if (colors[i] != colors[j]) {
                    res = Math.max(res, j - i);
                }
            }
        }
        return res;
    }
}